File: jpa-separate-ro-rw/src/test/java/com/sight/testType/IntegrationTest.java
package com.sight.testType;

import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.transaction.annotation.Transactional;

//@Transactional
@SpringBootTest
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
public abstract class IntegrationTest implements BaseTest { }

File: JPAwithRecursive/src/test/kotlin/com/sight/blog/config/test/IntegrationTest.kt
package com.sight.blog.config.test

import com.sight.blog.config.TestEnv
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.transaction.annotation.Transactional

@SpringBootTest
@ActiveProfiles(TestEnv.TEST)
@Transactional
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
abstract class IntegrationTest: BaseTest